db.go

```diff
@@ -588,8 +598,10 @@ func (db *DB) reload() (err error) {
 		}
 	}
 
-	if err := db.deleteBlocks(deletable); err != nil {
-		return err
+	if !db.opts.ReadOnly {
```

Comment:
All those `ifs` in this file. Do we really need to invoke `reload` and exactly the same `Open` functions in readOnly TSDB mode?

What readOnly mode should give? I think only tiny shim for Querying, right? Can we just add separate layer that does just that? It would massively reduce extra complexity in critical code here. What do you think? (: